### PR TITLE
Moved Sfm classes to their own files

### PR DESCRIPTION
**Pull request by dellaert**
- Moved SfmData and SfmTrack into sfm
- deprecated incorrectly named functions
- moved wrapping

### CODE REVIEW

File: gtsam/sfm/SfmTrack.h [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  /// @{
+
+  /// print
+  void print(const std::string& s = "") const;
+
+  /// assert equality up to a tolerance
+  bool equals(const SfmTrack& sfmTrack, double tol = 1e-9) const;
+
+  /// @}
+#ifdef GTSAM_ALLOW_DEPRECATED_SINCE_V42
```
Comment (johnwlambert) [inline gtsam/sfm/SfmTrack.h:98]: Hi Frank, thanks for the PR. Was curious how long we plan to keep this flag set to true by default (`GTSAM_ALLOW_DEPRECATED_SINCE_V42`).

We do use the `number_measurements()` flag and `number_tracks()` methods all over GTSFM, so we would need to refactor quite a few files. Was just curious the rationale for the renaming.

File: gtsam/sfm/SfmData.h [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file SfmData.h
+ * @date January 2022
+ * @author Frank dellaert
+ * @brief Data structure for dealing with Structure from Motion data
+ */
+
+#pragma once
```
Comment (johnwlambert) [inline gtsam/sfm/SfmData.h:19]: Good idea to split this up. The `slam/dataset.cpp` file was getting massive and was hard to follow.

File: gtsam/sfm/SfmTrack.h [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  explicit SfmTrack(const gtsam::Point3& pt, float r = 0, float g = 0,
+                    float b = 0)
+      : p(pt), r(r), g(g), b(b) {}
+
+  /// @}
+  /// @name Standard Interface
+  /// @{
+
+  /// Add measurement (camera_idx, Point2) to track
+  void addMeasurement(size_t idx, const gtsam::Point2& m) {
```
Comment (johnwlambert) [inline gtsam/sfm/SfmTrack.h:66]: this will be a breaking change too for GTSFM, which may be ok, but just something to be aware of (we are using `add_measurement()` currently.

File: gtsam/sfm/SfmTrack.h (hunk 1 above)
Comment (dellaert) [reply gtsam/sfm/SfmTrack.h:98]: Flag should actually be OFF by default.
Yes, I'm aware that if we build a new wheel (4.2a5) you'll have to amend GTSAM.

File: gtsam/sfm/SfmTrack.h (hunk 3 above)
Comment (dellaert) [reply gtsam/sfm/SfmTrack.h:66]: I'm aware

**Comment by johnwlambert** [comment]
Looks like CI is failing:
```
[ 24%] Building CXX object gtsam/CMakeFiles/gtsam.dir/sfm/SfmData.cpp.o
[ 24%] Building CXX object gtsam/CMakeFiles/gtsam.dir/sfm/SfmTrack.cpp.o
/home/runner/work/gtsam/gtsam/gtsam/sfm/SfmTrack.cpp:30:1: error: stray ‘\303’ in program
 bool SfmTrack::equals(const SfmTrack& sfmTrack, double tolÏ) const {
 ^
/home/runner/work/gtsam/gtsam/gtsam/sfm/SfmTrack.cpp:30:1: error: stray ‘\217’ in program
make[3]: *** [gtsam/CMakeFiles/gtsam.dir/sfm/SfmTrack.cpp.o] Error 1
gtsam/CMakeFiles/gtsam.dir/build.make:1797: recipe for target 'gtsam/CMakeFiles/gtsam.dir/sfm/SfmTrack.cpp.o' failed
make[3]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:3093: recipe for target 'gtsam/CMakeFiles/gtsam.dir/all' failed
make[2]: *** [gtsam/CMakeFiles/gtsam.dir/all] Error 2
CMakeFiles/Makefile2:2267: recipe for target 'CMakeFiles/check.dir/rule' failed
make[1]: *** [CMakeFiles/check.dir/rule] Error 2
Makefile:199: recipe for target 'check' failed
```

**Comment by dellaert** [comment]
> Looks like CI is failing:
>

I know, I'll fix it, along with addressing any other comments you might have. If you have no comments, please approve. If you do, *still approve* and trust me I will execute on them or talk to you if not :-)

File: gtsam/sfm/SfmData.cpp [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+Pose3 openGL2gtsam(const Rot3 &R, double tx, double ty, double tz) {
+  Rot3 R90 = openGLFixedRotation();
+  Rot3 wRc = (R.inverse()).compose(R90);
+
+  // Our camera-to-world translation wTc = -R'*t
+  return Pose3(wRc, R.unrotate(Point3(-tx, -ty, -tz)));
+}
+
+/* ************************************************************************* */
+Pose3 gtsam2openGL(const Rot3 &R, double tx, double ty, double tz) {
```
Comment (johnwlambert) [inline gtsam/sfm/SfmData.cpp:87]: Random comment on my end: I don't love the `x2y()` functions, since it makes me think x -> y, but from our Pose3 notation it could also mean x <- y. But out of the scope of this PR, so feel free to ignore.

**Comment by johnwlambert** [comment]
Sounds good @dellaert. The PR looks nice and I'll approve on my end. I like the decomposition of the BAL stuff into sfmData.cpp, since it never belonged in slamDataset.cpp so much.

I understand the snake case -> camel case change, to be consistent with our C++ guidelines. But I personally would find `SfmData.numberTracks()` to be more readable than `.nrTracks()`. This is a minor nit though, and if you feel like the latter is more consistent with GTSAM conventions, that's fine with me.

**Review by johnwlambert — APPROVED** [review]

File: gtsam/sfm/SfmData.cpp (hunk 4 above)
Comment (dellaert) [reply gtsam/sfm/SfmData.cpp:87]: I agree, but indeed out of scope

**Comment by dellaert** [comment]
> But I personally would find `SfmData.numberTracks()` to be more readable than `.nrTracks()`. This is a minor nit though, and if you feel like the latter is more consistent with GTSAM conventions, that's fine with me.

yeah, I could change that...
